client/core/core.go

```diff
@@ -5146,106 +5183,144 @@ func (c *Core) prepareTrackedTrade(dc *dexConnection, form *TradeForm, crypter e
-		c.notify(newOrderNote(TopicYoloPlaced, subject, details, db.Poke, corder))
-	} else {
-		rateString := "market"
-		if form.IsLimit {
-			rateString = wallets.trimmedConventionalRateString(corder.Rate)
+		dc.tradeMtx.Lock()
+		dc.trades[tracker.ID()] = tracker
+		dc.tradeMtx.Unlock()
+
+		// Now that the trades map is updated, sync with the preimage request.
+		c.syncOrderPlaced(ord.ID())
+
+		// Send a low-priority notification.
+		corder := tracker.coreOrder()
+		if !form.IsLimit && !form.Sell {
+			ui := wallets.quoteWallet.Info().UnitInfo
+			subject, details := c.formatDetails(TopicYoloPlaced,
+				ui.ConventionalString(corder.Qty), ui.Conventional.Unit, tracker.token())
+			c.notify(newOrderNoteWithTempID(TopicYoloPlaced, subject, details, db.Poke, corder, tempID))
+		} else {
+			rateString := "market"
+			if form.IsLimit {
+				rateString = wallets.trimmedConventionalRateString(corder.Rate)
+			}
+			ui := wallets.baseWallet.Info().UnitInfo
+			topic := TopicBuyOrderPlaced
+			if corder.Sell {
+				topic = TopicSellOrderPlaced
+			}
+			subject, details := c.formatDetails(topic, ui.ConventionalString(corder.Qty), ui.Conventional.Unit, rateString, tracker.token())
+			c.notify(newOrderNoteWithTempID(topic, subject, details, db.Poke, corder, tempID))
+		}
+
+		if isAccountRefund && fromWallet.parent != nil {
+			c.updateAssetBalance(fromWallet.parent.AssetID)
 		}
-		ui := wallets.baseWallet.Info().UnitInfo
-		topic := TopicBuyOrderPlaced
-		if corder.Sell {
-			topic = TopicSellOrderPlaced
+		if isAccountRedemption && toWallet.parent != nil {
+			c.updateAssetBalance(toWallet.parent.AssetID)
 		}
-		subject, details := c.formatDetails(topic, ui.ConventionalString(corder.Qty), ui.Conventional.Unit, rateString, tracker.token())
-		c.notify(newOrderNote(topic, subject, details, db.Poke, corder))
-	}
 
-	updated := assetMap{fromWallet.AssetID: struct{}{}}
-	if isAccountRefund && fromWallet.parent != nil {
-		updated[fromWallet.parent.AssetID] = struct{}{}
+		errCloser.Success()
+		return corder, nil
 	}
-	if isAccountRedemption && toWallet.parent != nil {
-		updated[toWallet.parent.AssetID] = struct{}{}
+
+	if !queue {
+		return submitAndValidateOrder(errorCloser, 0)
 	}
 
-	success = true
+	corder := coreOrderFromTrade(ord, meta)
+	asyncErrorCloser := errorCloser.Copy()
+	errorCloser.Success()
+	c.wg.Add(1)
+	go func() { // so core does not shut down while processing this order.
+		defer c.wg.Done()
+
+		// Store the inflight order and cleanup when it has been processed.
+		tempID := dc.storeInFlightOrder(corder)
+		defer dc.deleteInFlightOrder(tempID)
 
-	return corder, updated, nil
+		// Send silent note for the async order. This improves the UI/UX, so
+		// users don't have to wait for orders especially split tx orders.
+		topic := TopicAsyncOrderSubmitted
+		c.notify(newOrderNoteWithTempID(topic, "", "", db.Data, corder, tempID))
+
+		_, err := submitAndValidateOrder(asyncErrorCloser, tempID)
```

Comment:
I think you should get rid of both args to `submitAndValidateOrder`.  Declare `var tempID uint64` before defining the closure, and do `tempID = dc.storeInFlightOrder(corder)` (no `:=`) in the async branch, and just don't touch `tempID` in the synchronous branch.